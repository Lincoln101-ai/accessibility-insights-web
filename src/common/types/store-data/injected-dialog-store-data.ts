// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
export type InjectedDialogStoreData =
    | {
          isOpen: false;
          target: null;
      }
    | {
          isOpen: true;
          target: string[]; // Corresponds to the target property of an axe result node
      };
