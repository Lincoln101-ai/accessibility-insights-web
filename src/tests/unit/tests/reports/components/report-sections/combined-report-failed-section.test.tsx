// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { CollapsibleComponentCardsProps } from 'common/components/cards/collapsible-component-cards';
import { ScanMetadata } from 'common/types/store-data/unified-data-interface';
import { shallow } from 'enzyme';
import * as React from 'react';
import {
    CombinedReportFailedSectionProps,
    CombinedReportFailedSection,
    CombinedReportFailedSectionDeps,
} from 'reports/components/report-sections/combined-report-failed-section';
import { exampleUnifiedStatusResults } from 'tests/unit/tests/common/components/cards/sample-view-model-data';
import { It, Mock } from 'typemoq';

describe('CombinedReportFailedSection', () => {
    it('renders', () => {
        const collapsibleControlMock = Mock.ofType<
            (props: CollapsibleComponentCardsProps) => JSX.Element
        >();
        const scanMetaDataStub = {} as ScanMetadata;
        const containerId = 'container-id';
        const props: CombinedReportFailedSectionProps = {
            deps: {
                collapsibleControl: collapsibleControlMock.object,
            } as CombinedReportFailedSectionDeps,
            containerId,
            content: <div>Content</div>,
            cardsViewData: {
                cards: exampleUnifiedStatusResults,
                visualHelperEnabled: true,
                allCardsCollapsed: true,
            },
            scanMetadata: scanMetaDataStub,
        } as CombinedReportFailedSectionProps;

        const expectedCollapsibleControlProps: Partial<CollapsibleComponentCardsProps> = {
            headingLevel: 2,
            deps: null,
        };

        collapsibleControlMock
            .setup(cc => cc(It.isObjectWith(expectedCollapsibleControlProps)))
            .returns(() => <div>Collapsible component</div>);

        const wrapper = shallow(<CombinedReportFailedSection {...props} />);
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
