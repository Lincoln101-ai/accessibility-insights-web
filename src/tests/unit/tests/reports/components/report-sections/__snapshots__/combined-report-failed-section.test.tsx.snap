// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CombinedReportFailedSection renders 1`] = `
"<div className=\\"result-section\\">
  <div id=\\"collapsible-header\\">
    <CombinedResultSectionTitle outcomeCount={1} outcomeType=\\"fail\\" title=\\"Failed\\" titleSize=\\"title\\" deps={{...}} containerId=\\"container-id\\" content={{...}} cardsViewData={{...}} scanMetadata={{...}} />
  </div>
  <div id=\\"collapsible-content\\">
    <ResultSectionContent outcomeType=\\"fail\\" targetAppInfo={[undefined]} results={{...}} visualHelperEnabled={true} allCardsCollapsed={true} userConfigurationStoreData={{...}} outcomeCounter={[Function: countByIdentifierUrls]} deps={{...}} containerId=\\"container-id\\" content={{...}} cardsViewData={{...}} scanMetadata={{...}} />
  </div>
</div>"
`;
